borg/fuse.py

```diff
@@ -76,6 +76,16 @@ def process_archive(self, archive, prefix=[]):
             data = self.key.decrypt(key, chunk)
             unpacker.feed(data)
             for item in unpacker:
+                try:
+                    # This can happen when an archive was created with a command line like
+                    # $ borg create ... dir1/file dir1
+                    # In this case the code below will have created a default_dir inode for dir1 already.
+                    inode = self._find_inode(item[b'path'], prefix)
+                except KeyError:
+                    pass
+                else:
+                    self.items[inode] = item
```

Comment:
I think the intention is that all archive-items go into the cache, where normally also directories would end up. get_item() is used to access that, which also checks self.items before accessing the cache, so it just works fine. There' s `self.cache.get(inode)` call below, but that handles hardlinks (b'source' in item and ISREG), so that inode can't point to this kind of directory inode, therefore the call is safe.